src/redux/missions/missionsSlice.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable implicit-arrow-linebreak */
+/* eslint-disable comma-dangle */
+/* eslint-disable function-paren-newline */
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const FETCH_MISSIONS = 'missions/fetchMissions';
+const url = 'https://api.spacexdata.com/v3/missions/';
+const fetchMissions = createAsyncThunk(FETCH_MISSIONS, () =>
+  axios.get(url).then((response) => {
+    const missions = response.data;
+    const cleanData = {};
+    missions.forEach((element) => {
+      const { mission_id: id, mission_name: name, description } = element;
+      cleanData[id] = {
+        id,
+        name,
+        description,
+        reserved: false,
+      };
+    });
+    return cleanData;
+  })
+);
+
+export { fetchMissions };
+
+const initialState = {
```

Comment:
- [ ] Kindly fix the error which occurred when the mission is added when we return back to the mission page after checking my profile page the selected option from the mission disappears. Have a look at the below video to understand better.

https://user-images.githubusercontent.com/70269507/203843778-f20013a5-dc9e-413d-9156-92d237917a7b.mov